src/main/java/com/google/sps/servlets/EditBusinessDataServlet.java

```diff
@@ -0,0 +1,248 @@
+        Entity businessEntity = getBusinessEntity(datastore, businessKey);
+
+        String name = (String) businessEntity.getProperty(BUSINESS_NAME);
+        String desc = (String) businessEntity.getProperty(BUSINESS_DESC);
+        String[] categoriesArr = gson.fromJson((String) businessEntity.getProperty(BUSINESS_CATEGORIES), String[].class);
+        List<String> categories = Arrays.asList(categoriesArr);
+        String address = (String) businessEntity.getProperty(BUSINESS_ADDRESS);
+        float addressLat = ((Double) businessEntity.getProperty(BUSINESS_ADDRESS_LAT)).floatValue();
+        float addressLng = ((Double) businessEntity.getProperty(BUSINESS_ADDRESS_LNG)).floatValue();
+        String contactDetails = (String) businessEntity.getProperty(BUSINESS_CONTACT_INFO);
+        String orderDetails = (String) businessEntity.getProperty(BUSINESS_ORDER_INFO);
+        float minPrice = ((Double) businessEntity.getProperty(BUSINESS_MIN_PRICE)).floatValue();
+        float maxPrice = ((Double) businessEntity.getProperty(BUSINESS_MAX_PRICE)).floatValue();
+        String businessLink = (String) businessEntity.getProperty(BUSINESS_LINK);
+        String menuLink = (String) businessEntity.getProperty(BUSINESS_MENU_LINK);
+        String logoUrl = (String) businessEntity.getProperty(BUSINESS_LOGO);
+        String[] picturesUrlsArr = gson.fromJson((String) businessEntity.getProperty(BUSINESS_PICTURES), String[].class);
+        List<String> picturesUrls = picturesUrlsArr == null ? new ArrayList<String>() : Arrays.asList(picturesUrlsArr);
+        float rating = ((Double) businessEntity.getProperty(BUSINESS_RATING)).floatValue();
+        String[] reviewsArr = gson.fromJson((String) businessEntity.getProperty(BUSINESS_REVIEWS), String[].class);
+        List<String> reviews = reviewsArr == null ? new ArrayList<String>() : Arrays.asList(reviewsArr);
+
+
+        Business business = new Business(name, categories, minPrice, maxPrice, rating, addressLat, addressLng, address, logoUrl, picturesUrls, desc, menuLink, orderDetails, contactDetails, businessLink);
+        String businessJson = String.format("{\"data\" : %s, \"id\": %s }", gson.toJson(business), businessEntity.getKey().getId());
+        // Send the JSON as the response.
+        response.setContentType("application/json;");
+        response.getWriter().println(businessJson);
+    } catch (IOException err) {
+        System.out.println(err);
+    } catch (EntityNotFoundException err) {
+        System.out.println(err);
+    }
+  }
+
+
+  /** Gets input from the form, including the business key, retrieves the business matching the key
+  * and updates it with the fields from the form.
+  */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson gson = new Gson();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Consider initializing this with the Gson object to prevent us create a DataStoreService every POST call.

Reply:
oh you mean as a class variable? like how Lilian did it in her servlet? (If you can remember haha)